src/Microsoft.Azure.SignalR.Common/ServiceConnections/StrongServiceConnectionContainer.cs

```diff
@@ -20,11 +20,14 @@ internal class StrongServiceConnectionContainer : ServiceConnectionContainerBase
         public override async Task HandlePingAsync(PingMessage pingMessage)
         {
             await base.HandlePingAsync(pingMessage);
-            if (RuntimeServicePingMessage.TryGetRebalance(pingMessage, out var target) && !string.IsNullOrEmpty(target))
+            if (ReadyForNewConnections)
```

Comment:
instead of adding the check here, we should let RT not sending rebalance request when it's going offline.

Reply:
Should be both - SDK not creating new connections when in shutting down stage and RT not sending rebalance requests. Since there are inevitable (and sometimes unpredictable) delays of incoming messages the actual rebalance ping could have been sent before the shutdown started.